### PR TITLE
Updated documentation

### PR DESCRIPTION
**Pull request by glower**
* Added documentation about the frontend build process
* Added documentation about the contribution and dependencies management
* Updated version of Go for docker

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -29,10 +29,10 @@ in order to changes take effect. That could require a browser restart.
 NEVER set this configuration in production environments.
 
 ## Install from Source
-If you have a Go development environment set up, Go get it:
+If you have a Go development environment set up, Go get it, we recommend at least golang v1.13
```
Comment (suligap) [inline README.md]: Maybe ". We recommend golang v1.13 or higher." instead?

File: README.md [hunk 2]
```diff
@@ -70,6 +94,28 @@ everything apart from the Apache front-end units. There is an example of using J
 
 ## Development Environment
 
+### Contributing
+
+The general workflow is forking the Serial Vault GitHub repository, make changes in a branch and then send a pull request:
```
Comment (suligap) [inline README.md]: s/send/create/ ?

File: README.md [hunk 3]
```diff
@@ -70,6 +94,28 @@ everything apart from the Apache front-end units. There is an example of using J
...
+  `go get github.com/CanonicalLtd/serial-vault/...`
+- [Fork](https://github.com/CanonicalLtd/serial-vault/fork) the Serial Vault repository on Github
+- Change to the top level of the repository
+  `cd $GOPATH/src/github.com/CanonicalLtd/serial-vault`
+- Add your fork
+  `git remote add fork https://github.com/username/serial-vault`
+- Create a feature branch
+  `git checkout -b new-feature`
+- Commit your changes to your forked repo
+  `git commit -am "New Feature"`
```
Comment (suligap) [inline README.md]: Maybe separate the two commands with some text, like " and "?

File: README.md [hunk 4]
```diff
@@ -108,14 +154,20 @@ npm install
...
 ```bash
 # Select the version to use
+cd webapp-admin/
 nvm ls
 nvm use lts/*
 npm run build
 ```
 
+#### Build the project bundle for the production
+Production build for the frontend part (javascript and css) is semi-automated and done with [github actions](https://github.com/features/actions). You can find the configuration for this process in `.github/workflows/nodejs.yml`. The build process starts automatically after the PR is approved and pushed to `master`. You can see the build process in [actions](https://github.com/CanonicalLtd/serial-vault/actions) tab of this project. After the successful build the automation bot will create a PR with the generated build artifact (minified javascript code) in the `static/` directory of this project. This PR should be approved manually by the developer.
```
Comment (suligap) [inline README.md]: Maybe we don't need the last sentence? ("This PR should be approved manually by the developer.")

Or change it to: "These PRs can be merged manually."?

File: README.md [hunk 5]
```diff
@@ -108,14 +154,20 @@ npm install
...
+#### Build the project bundle for the local development
 ```bash
 # Select the version to use
+cd webapp-admin/
 nvm ls
 nvm use lts/*
 npm run build
 ```
 
+#### Build the project bundle for the production
```
Comment (suligap) [inline README.md]: "Production static assets build process"?

File: README.md [hunk 6]
```diff
@@ -108,14 +154,20 @@ npm install
 
 ### Working with React
 
-#### Build the project bundle
+The frontend code could be found in `webapp-admin` directory.
+
+#### Build the project bundle for the local development
```
Comment (suligap) [inline README.md]: "Building static assets locally"?

File: README.md [hunk 7]
```diff
@@ -70,6 +94,28 @@ everything apart from the Apache front-end units. There is an example of using J
 
 ## Development Environment
 
+### Contributing
```
Comment (doismellburning) [inline README.md]: This feels like general GitHub "filing a PR" docs - do they have something we can just refer to from here?

File: README.md (hunk 4 above)
Comment (doismellburning) [inline README.md]: "GitHub Actions" not "github actions"

Also thoughts on [semantic linebreaks](https://blog.doismellburning.co.uk/semantic-linebreaks/)? E.g. this long line doesn't play nicely with GitHub's review UI

File: README.md (hunk 1 above)
Comment (doismellburning) [reply README.md]: Also I believe it's always "Go" and not "golang", outside of some specific contexts (I can't find the source I saw, but you'll notice this in e.g. https://github.com/golang/go/wiki)

File: README.md [hunk 8]
```diff
@@ -70,6 +94,28 @@ everything apart from the Apache front-end units. There is an example of using J
...
+
+The general workflow is forking the Serial Vault GitHub repository, make changes in a branch and then send a pull request:
+
+- Pull the original package:
+  `go get github.com/CanonicalLtd/serial-vault/...`
+- [Fork](https://github.com/CanonicalLtd/serial-vault/fork) the Serial Vault repository on Github
```
Comment (doismellburning) [inline README.md]: GitHub not Github

File: README.md [hunk 9]
```diff
@@ -108,14 +154,20 @@ npm install
 
 ### Working with React
 
-#### Build the project bundle
+The frontend code could be found in `webapp-admin` directory.
```
Comment (doismellburning) [inline README.md]: "can", not "could"

File: README.md (hunk 1 above)
Comment (glower) [reply README.md]: yes, make sense

File: README.md (hunk 7 above)
Comment (glower) [reply README.md]: I have a look if I can find something in the GitHub documentation I can refer

**Comment by doismellburning** [comment]
(I assume you plan to rebase-squash before merge?)

**Pull request merged**